cert/cert.go

```diff
@@ -120,25 +130,51 @@ func (ca *CA) Refresh() (bool, error) {
 		return false, err
 	}
 
-	if bytes.Equal(cert, ca.pem) {
+	isCACertSame, err := CompareCertificates(cert, ca.pem)
+	if err != nil {
+		log.Infof("cert: error comparing CA certificates")
+		return false, err
+	}
+
+	if isCACertSame {
 		if ca.File != nil {
 			log.Infof("cert: existing CA certificate at %s is current", ca.File.Path)
 		}
 		return false, nil
 	}
 
+	// If CA cert has changed, write out new CA cert
 	if ca.File != nil {
 		err = ca.writeCert(cert)
 	}
 	// If there were no errors, update our internal notion of what the CA is.
 	if err != nil {
-		// see CA.Load(); strip prefix/trailing whitespace to ensure our bytes.Equal() checks don't false positive.
-		ca.pem = []byte(strings.TrimSpace(string(cert[:])))
+		ca.pem = cert
 	}
-
 	return true, err
 }
 
+// CompareCertificates x509 compares two CA certificates
+func CompareCertificates(cert1, cert2 []byte) (bool, error) {
```

Comment:
My rough thought would be to hold the x509 structure, rather than having a pem []byte floating around that we reparse every run.  I'm guessing the encapsulation makes this a pain in the ass to internalize?

Reply:
Yeah pretty much... holding the x509 structure would require changing the CA struct (and all references to it) as well as the getRemoteCert() function. I made the inputs here two []byte because that's what's readily available in the CA struct & getRemoteCert() function. 

At least this function gets rid of the gnarly byte comparison logic from before, though! The unit tests verify that trailing newlines shouldn't be a problem anymore.